File: app/src/main/java/workshop1024/com/xproject/model/news/source/local/NewsDao.java
package workshop1024.com.xproject.model.news.source.local;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Query;

import java.util.List;

import workshop1024.com.xproject.model.news.News;

@Dao
public interface NewsDao {
    @Query("SELECT * FROM news")
    List<News> getNewses();

    @Query("SELECT * FROM news WHERE publisherId = :publisherId")
    List<News> getNewsByPublisherId(String publisherId);
}

File: news/src/main/java/workshop1024/com/xproject/news/model/news/local/NewsDao.kt
package workshop1024.com.xproject.news.model.news.local

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import workshop1024.com.xproject.news.model.news.News

@Dao
interface NewsDao {

    @Query("SELECT * FROM newses WHERE searchType = :searchType AND searchKey = :searchKey")
    fun getNewsesByTypeAndKey(searchType: String, searchKey: String): List<News>

    @Query("DELETE FROM newses WHERE searchType = :searchType AND searchKey = :searchKey")
    fun deleteAllNewsesByTypeAndKey(searchType: String, searchKey: String)

    @Query("UPDATE newses SET isReaded = 1 WHERE newsId = :newsId")
    fun markNewsReadedByNewsId(newsId: String)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addNews(news: News)
}